convert2rhel/checks.py

```diff
@@ -243,11 +243,14 @@ def get_rhel_supported_kmods():
     # get output of a command to get all packages which are the source
     # of kmods
     kmod_pkgs_str, _ = run_subprocess(
-        ("repoquery --releasever={releasever} {setopt_arg} {repoids_args} -f /lib/modules/*.ko*").format(
-            releasever=system_info.releasever,
-            setopt_arg=setopt_arg,
-            repoids_args=repoquery_repoids_args,
-        ),
+        [
+            "repoquery",
+            "--releasever=%s" % system_info.releasever,
+            setopt_arg,
+            repoquery_repoids_args,
```

Comment:
I think setopt_arg and reppoquery_repoids_args here need to be modified a bit:
* setopt_arg might be set to the empty string: https://github.com/oamg/convert2rhel/pull/389/files#diff-cf19bc90f65d9e26e1b6fa7dcd84a18dfba097f65706e211a4f4ac71b1b467f6R242  I'm not sure if repoquery will ignore that or attempt to do something with it that fails.  It would be better to only add setopt_arg if it exists.
* repoquery_repoids_args is being set to a string with spaces for each repository like this:
  `'--repoid fedora --repoid fedora-updates'`.  We need each of those to be separate entries in the list.

Given those constraints, I think we need to build up the command as a list outside of the run_subprocess() rather than trying to construct the list inside of the function call.  Something like we did in pkghandler: https://github.com/oamg/convert2rhel/pull/389/files#diff-361f5dc14483ec43e37529967acee396da5714dbf6562472ccece5db4bba5e0dR134

Reply:
@Andrew-ang9 Could you take a look at fixing this one?

Reply:
Yeah I can take care of that!